assets/src/components/notificationCard.tsx

```diff
@@ -43,52 +57,117 @@ export const NotificationCard = ({
-          fullStoryEvent("User clicked Chelsea Bridge Notification", {})
-        }
-      }}
-      closeCallback={hideLatestNotification}
-      time={notification.createdAt}
-      noFocusOrHover={noFocusOrHover}
-    >
-      <CardBody>{description(notification, routes, routeAtCreation)}</CardBody>
-      {isBlockWaiverNotification(notification) && (
-        <CardProperties
-          properties={[
-            {
-              label: "Run",
-              value:
-                notification.content.runIds.length > 0
-                  ? notification.content.runIds.join(", ")
-                  : null,
-            },
-            {
-              label: "Operator",
-              value:
-                notification.content.operatorName !== null &&
-                notification.content.operatorId !== null
-                  ? `${notification.content.operatorName} #${notification.content.operatorId}`
-                  : null,
-              sensitive: true,
-            },
-          ]}
+    <>
+      <CardReadable
+        currentTime={currentTime}
+        title={<>{title(notification)}</>}
+        style="kiwi"
+        isActive={isUnread}
+        openCallback={() => {
+          if (notification.content.$type === NotificationType.Detour) {
+            setShowDetourModal(true)
+          } else {
+            openVPPForCurrentVehicle(notification)
+
+            if (hideLatestNotification) {
+              hideLatestNotification()
+            }
+
+            if (
+              notification.content.$type === NotificationType.BridgeMovement
+            ) {
+              fullStoryEvent("User clicked Chelsea Bridge Notification", {})
+            }
+          }
+        }}
+        closeCallback={hideLatestNotification}
+        time={notification.createdAt}
+        noFocusOrHover={noFocusOrHover}
+      >
+        <CardBody>
+          {description(notification, routes, routeAtCreation)}
+        </CardBody>
+        {isBlockWaiverNotification(notification) && (
+          <CardProperties
+            properties={[
+              {
+                label: "Run",
+                value:
+                  notification.content.runIds.length > 0
+                    ? notification.content.runIds.join(", ")
+                    : null,
+              },
+              {
+                label: "Operator",
+                value:
+                  notification.content.operatorName !== null &&
+                  notification.content.operatorId !== null
+                    ? `${notification.content.operatorName} #${notification.content.operatorId}`
+                    : null,
+                sensitive: true,
+              },
+            ]}
+          />
+        )}
+      </CardReadable>
+      {detourId && (
+        <DetourNotificationModal
+          show={showDetourModal}
+          detourId={detourId}
+          onClose={onCloseDetour}
         />
       )}
```

Comment:
Hmm, I think we should gate this also on `showDetourModal`? because otherwise I think this is going to fetch and load every detour in the background whenever there's a notification?
```suggestion
      {detourId && showDetourModal && (
        <DetourNotificationModal
          show={showDetourModal}
          detourId={detourId}
          onClose={onCloseDetour}
        />
      )}
```

Reply:
this _miiiiight_ fix the test too.

Reply:
Aha, thank you!! This was needed, as was a conditional around the <DetourModal/> to ensure the fetch for a detour returned before rendering.